akka-cluster-sharding-typed/src/test/scala/docs/akka/cluster/sharding/typed/ShardingCompileOnlySpec.scala

```diff
@@ -100,4 +100,30 @@ object ShardingCompileOnlySpec {
       .withStopMessage(GoodByeCounter))
   //#counter-passivate
 
+  def counterWithResponseToShardedActor(): Unit = {
+
+    //#sharded-response
+    // a sharded actor that needs counter updates
+    trait Command
+    final case class NewCount(count: Long) extends Command
+    val entityTypeKey: EntityTypeKey[Command] = EntityTypeKey[Command]("example-sharded-response")
+
+    // a sharded counter that sends responses to another sharded actor
+    trait CounterCommand
+    case object Increment extends CounterCommand
+    final case class GetValue(replyToEntityId: String) extends CounterCommand
+
+    def counter(value: Long): Behavior[CounterCommand] =
+      Behaviors.receiveMessage[CounterCommand] {
+        case Increment =>
+          counter(value + 1)
+        case GetValue(replyToEntityId) =>
+          sharding.entityRefFor(entityTypeKey, replyToEntityId) ! NewCount(value)
```

Comment:
maybe an alternative could be to include the `EntityTypeKey` in the `GetValue` message. Should be ok to serialize as String of the `EntityTypeKey.name`

Reply:
Could we even send the whole `EntityRef`? Or would that cause trouble in serialization?

Reply:
It’s typically not sent standalone but enclosed in another message, and the serializer for that would need to know how to serialize the ref and resolve it on the other side. Boils down to a string which is the entityId and resolved with sharding.entityRefFor(id). This is not something we can automate (well, for Jackson we could).

Reply:
Right - that's the same with including the `ActorRef` in the message as `replyTo` argument, though, I think?

Reply:
It contains a reference to the local shard region which makes it not transferable between nodes so I think it is not serializable by design

Reply:
anyhow, let's not do that in this PR

it could be possible to transfer it as `entityTypeKey: String` + `entityId: String` and then use `ClusterSharding(system).entityRefFor(...)` when deserializing (modulo types which are just casted in same way as when deserializing a typed ActorRef). However, I think I still prefer to let that be an application level concern than hiding it in serialization.